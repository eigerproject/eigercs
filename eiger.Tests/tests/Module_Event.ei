include event

let ev = Event()
ev.Subscribe(func(msg) emitln("First Subscriber: " + msg) end)
ev.Subscribe(func(msg) emitln("Second Subscriber: " + msg) end)
ev.Invoke("Hello!")

func functionOne(args) > emitln("This is function one")
func functionTwo(args) > emitln("This is function two")
ev = Event()
ev.Subscribe(functionTwo)
ev.Subscribe(functionOne)
ev.Unsubscribe(functionTwo)
ev.Invoke(nix)