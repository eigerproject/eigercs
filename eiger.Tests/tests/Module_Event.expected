First Subscriber: Hello!
Second Subscriber: Hello!
This is function one